tests/utils/helpers.rs

```diff
@@ -1187,12 +1191,43 @@ impl TestWallet {
             .unwrap()
     }
 
+    pub fn psbt_add_input(&mut self, psbt: &mut Psbt, utxo: Outpoint) {
+        for account in self.descriptor.xpubs() {
+            psbt.xpubs.insert(*account.xpub(), account.origin().clone());
+        }
+        let input = self.wallet.wallet().utxo(utxo).unwrap();
+        psbt.construct_input_expect(
+            input.to_prevout(),
+            self.wallet.wallet().descriptor(),
+            input.terminal,
+            SeqNo::ZERO,
+        );
+    }
+
     pub fn color_psbt(
         &mut self,
         psbt: &mut Psbt,
         coloring_info: ColoringInfo,
     ) -> (Fascia, AssetBeneficiariesMap) {
-        let _output = psbt.construct_output_expect(ScriptPubkey::op_return(&[]), Sats::ZERO);
+        let asset_beneficiaries = self.color_psbt_init(psbt, coloring_info);
+        psbt.complete_construction();
+        let fascia = psbt.rgb_commit().unwrap();
+        (fascia, asset_beneficiaries)
+    }
+
+    pub fn color_psbt_init(
+        &mut self,
+        psbt: &mut Psbt,
+        coloring_info: ColoringInfo,
+    ) -> AssetBeneficiariesMap {
+        if !psbt
+            .to_unsigned_tx()
+            .outputs
+            .iter()
+            .any(|o| o.script_pubkey.is_op_return())
```

Comment:
PSBT `Outputs` contain script pubkey, so no need for the allocation-expensive `to_unsigned_tx`. You can do something like `if !psbt.outputs().any(|o| ...)`

Reply:
Fixed, thanks for spotting this!